app/experimenter/experiments/tests/test_forms.py

```diff
@@ -202,6 +205,302 @@ class Meta:
+                },
+            ],
+            latest_changes.new_values["variants"],
+        )
+
+    def test_mixin_sets_with_m2m_with_existing_changelog(self):
+        experiment = ExperimentFactory.create_with_variants(num_variants=0)
+
+        ExperimentChangeLog.objects.create(
+            experiment=experiment,
+            changed_by=self.request.user,
+            old_status=Experiment.STATUS_ACCEPTED,
+            new_status=Experiment.STATUS_DRAFT,
+            old_values={"variants": None},
+            new_values={
+                "variants": [
+                    {
+                        "name": " old branch 1 name",
+                        "slug": " old branch-1-name",
+                        "ratio": 50,
+                        "value": "8",
+                        "is_control": True,
+                        "description": " old branch 1 desc",
+                    }
+                ]
+            },
+            message="",
+        )
+
+        experiment.save()
+
+        data = {
+            "population_percent": "10",
+            "firefox_version": "56.0",
+            "firefox_channel": Experiment.CHANNEL_BETA,
+            "client_matching": "en-us",
+            "platform": Experiment.PLATFORM_WINDOWS,
+            "locales": [],
+            "countries": [],
+            "pref_key": "some pref key",
+            "pref_type": Experiment.PREF_TYPE_INT,
+            "pref_branch": Experiment.PREF_BRANCH_DEFAULT,
+            "addon_experiment_id": "add_on_experiment_id",
+            "addon_release_url": "https://www.example.com",
+            "variants-TOTAL_FORMS": "2",
+            "variants-INITIAL_FORMS": "0",
+            "variants-MIN_NUM_FORMS": "0",
+            "variants-MAX_NUM_FORMS": "1000",
+            "variants-0-is_control": True,
+            "variants-0-ratio": "50",
+            "variants-0-name": "variant 0 name",
+            "variants-0-description": "variant 0 desc",
+            "variants-0-value": 5,
+            "variants-1-is_control": False,
+            "variants-1-ratio": "50",
+            "variants-1-name": "branch 1 name",
+            "variants-1-description": "branch 1 desc",
+            "variants-1-value": 8,
+        }
+
+        form = ExperimentVariantsPrefForm(
+            request=self.request, data=data, instance=experiment
+        )
+        self.assertTrue(form.is_valid())
+        experiment = form.save()
+        latest_changes = experiment.changes.latest()
+        self.assertCountEqual(
+            [
+                {
+                    "name": " old branch 1 name",
+                    "slug": " old branch-1-name",
+                    "ratio": 50,
+                    "value": "8",
+                    "is_control": True,
+                    "description": " old branch 1 desc",
+                }
+            ],
+            latest_changes.old_values["variants"],
+        )
+
+        self.assertEquals(len(latest_changes.new_values["countries"]), 0)
```

Comment:
Ah I see, this tests wiping out the countries/locales, but should we be testing the case where we start with some set countries/locales, and wind up with a modified set of locales/countries (ie adding some, removing some)?